### ISSUE TITLE
training results

### ISSUE THREAD

**Issue by sc-56** [xu-ji/IIC#74]
Sorry for bothering again, 

I notice there are 5 results in each training epoch, whose format is just like
train_accs: [0.9918857, 0.97852856, 0.9005857, 0.97854286, 0.97852856] 

May I roughly know what's the difference of those results? I experimented to change the tf2, and some of the results would increase, so are they assoicated with the transformation format?

Thank you so much for your help.

**Comment by xu-ji** [comment]
That's the list of accuracies on the data for each of the 5 sub heads. If you set --num_sub_heads 1 there should be 1 result.

The data that produced that accuracy is the data in `mapping_assignment_dataloader`, i.e. the labelled part of the training data.

**Comment by sc-56** [comment]
Thank you so much for your explanation.

**Comment by sc-56** [comment]
Sorry for bothering again,

I use the '685' command and original script to train the classifer for the MNIST, and one thing I'm a little confused, which is  sometimes I can get the similar result with the 'training log file'. However, sometimes the result seems to be not increasing in one number for long time (epoch after epoch the number seems not grow too much).

ex. 
pre-epoch: [0.8790857, 0.87932855, 0.8792286, 0.87934285, 0.8789857]
next:           [0.8793714, 0.87897146, 0.8791143, 0.8788143, 0.8786]

Sometimes (1-3) of the 5 sub-heads are outstanding than others, sometimes 5 sub-heads are normal. Therefore, if i wish to produce better performance model, should I try severial times to find out best?

Thank you for your support.

**Comment by xu-ji** [comment]
Depending on your needs, you could indeed try several times and then select the "best" sub-head as the one with the lowest loss (`--select_sub_head_on_loss`), or highest accuracy as you have been doing. For a benchmark result it's good to report mean and standard deviation (I regret not doing this in the paper).

If you are doing a fully unsupervised experiment, choose best sub-head based on lowest loss (to avoid using labels for the sub-head selection). You could also use [a script](https://github.com/xu-ji/IIC/blob/master/examples/commands.txt#L115) to do this after saving a model.

**Comment by sc-56** [comment]
Thank you so much for your explanation.
